### ISSUE TITLE
When the Godot editor is fullscreened on a Mac, Context Menus don't show up until you move the mouse 

### ISSUE THREAD

**Issue by llanes1990** [godotengine/godot#81828]
### Godot version

v4.1.1.stable.mono.official [bd6af8e0e]

### System information

Macbook Pro 14-inch - M2 Max - MacOS Ventura 13.5.1

### Issue description

When Godot is fullscreened/maximized and I right-click to open a context menu, the context menu doesnt immediately show until I move the mouse. I notice a similar issue when opening dropdown boxes or dialog boxes.

I notice that the issue goes away when one of the following happens:
1. Godot is un-fullscreened/unmaximized
2. A screen recording is happening (Cmd+shift+5 and Record the screen/selected portion)

### Steps to reproduce

Open Godot on a Mac, leave it non-full-screened, and open a project.

While non-full-screened:
1. Right-click somewhere to open a context menu (Do not move the mouse after.)
2. Notice that the context-menu immediately appears

Maximize/Fullscreen the editor:
1. Right-click somewhere to open a context menu. (Do not move the mouse after.)
2. Notice that the context-menu does not appear regardless of how long you wait.
3. Move the mouse.
4. Notice that the context-menu appears.

### Minimal reproduction project

N/A. Issue is reproducible on a new empty project.

**Comment by jalder89** [comment]
Confirmed reproducible in these Godot Engine builds:
- v4.2.1.stable.official [b09f793f5]
- v4.2.1.stable.mono.official [b09f793f5]
- v4.3.dev1.official [9d1cbab1c]
- v4.3.dev1.mono.official [9d1cbab1c]

Reproduced on:
- Vulkan API 1.2.231 - Forward+ - Using Vulkan Device #0: Apple - Apple M3 Max
- MacOS Sonoma 14.2.1

Moving the mouse will trigger the context menu to open as described. However, I noticed doing so also causes the close/min/max buttons and system menus for Godot to disappear until focus is moved to another app. For example, moving to Chrome and clicking inside the window and then moving back to fullscreen Godot restores the close/min/max buttons.

I created a second issue for the system menus and controls, but they may have the same root cause as the failing editor menus: #86480
